test/unit/vl-http-error-message.test.html

```diff
@@ -25,84 +25,165 @@
+    <template>
+      <vl-http-500-message></vl-http-500-message>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="vl-http-500-message-slot-fixture">
+    <template>
+      <vl-http-500-message>
+        <div slot="actions">
+          <span>Knop</span>
+        </div>
+      </vl-http-500-message>
+    </template>
+  </test-fixture>
+
   <script type="module">
     import { awaitUntil } from 'vl-ui-core/dist/vl-core';
+    const sandbox = sinon.sandbox.create();
+
+    teardown(() => {
+      sandbox.restore();
+    });
+
+    const assertWithElements = (fixture, assert) => {
+      const message = fixture;
+      const shadowRoot = message.shadowRoot;
+      const title = shadowRoot.querySelector('#title');
+      const text = shadowRoot.querySelector('#text');
+      const actions = shadowRoot.querySelector('#actions');
+      const action = shadowRoot.querySelector('#actions > *');
+      const smallImage = shadowRoot.querySelector('#image-small');
+      const normalImage = shadowRoot.querySelector('#image-normal');
+      assert({
+        message: message,
+        title: title,
+        text: text,
+        actions: actions,
+        action: action,
+        smallImage: smallImage,
+        normalImage: normalImage
+      });
+    };
+
+    const assertConfig = (fixture) => {
+      const message = fixture;
+      message.dataset.vlTitle = 'Foo';
+      message.dataset.vlImage = 'Bar';
+      message.dataset.vlImageAlt = 'foobar';
+
+      const textSlot = document.createElement('div');
+      textSlot.slot = 'text';
+      textSlot.textContent = 'hello world';
+      message.appendChild(textSlot);
+
+      const actionsSlot = document.createElement('div');
+      actionsSlot.slot = 'actions';
+      actionsSlot.textContent = 'moo';
+      message.appendChild(actionsSlot);
+
+      const shadowRoot = message.shadowRoot;
+      const smallImage = shadowRoot.querySelector('#image-small');
+      const normalImage = shadowRoot.querySelector('#image-normal');
+
+      return awaitUntil(() => {
+        return shadowRoot.querySelector('#title').textContent === 'Foo' &&
+          shadowRoot.querySelector('#text').textContent === 'hello world' &&
+          shadowRoot.querySelector('#actions').textContent === 'moo' &&
+          smallImage.getAttribute('src') === 'Bar' &&
+          smallImage.alt === 'foobar' &&
+          normalImage.getAttribute('src') === 'Bar' &&
+          normalImage.alt === 'foobar'
+      });
+    }
 
     suite('vl-http-error-message', () => {
       const should = chai.should();
 
-      test('Een vl-http-error-message toont alle geconfigureerde tekst, afbeeldingen en acties', () => {
-        const message = fixture('vl-http-error-message-fixture');
-        const shadowRoot = message.shadowRoot;
+      test('toont alle geconfigureerde tekst, afbeeldingen en acties', () => {
+        assertWithElements(fixture('vl-http-error-message-fixture'), ({ title, text, actions, smallImage, normalImage }) => {
```

Comment:
ik ben niet zo voor hulpmethodes in testen :-)

Reply:
Ik ook niet, maar ik heb de stijl van de persoon die deze webcomponent initieel geïmplementeerd heeft overgenomen.

Reply:
twas een mopje hoor :-)